### PR TITLE
jira - some improvements to the module

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
- Fixed examples documentation.
- Module no longer incorrectly reports change for information gathering operations.
- Replaced custom parameter validation with ``required_if``.
- Added the ``traceback`` output to ``fail_json()`` calls deriving from exceptions.

This work was prompted by looking at #1233 but I am not sure whether it solves the issue.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
jira

### CODE REVIEW

File: changelogs/fragments/jira_improvements.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+bugfixes:
+  - Fixed examples documentation ().
+  - Module no longer incorrectly reports change for information gathering operations ().
+  - Replaced custom parameter validation with ``required_if`` ().
+minor_changes:
+  - Added the traceback output to ``fail_json()`` calls deriving from exceptions ().
```
Comment (felixfontein) [inline changelogs/fragments/jira_improvements.yaml:6]: This file needs an update :)

Comment (russoz) [reply changelogs/fragments/jira_improvements.yaml]: Yes, as there was no issue # I was waiting for the PR #, and then I started another bugfix and forgot that. Just pushed an update.

File: plugins/modules/web_infrastructure/jira.py [hunk 2]
```diff
@@ -500,19 +487,19 @@ def main():
             timeout=dict(type='float', default=10),
             validate_certs=dict(default=True, type='bool'),
         ),
+        required_if=(
+            ('operation', 'create', ['project', 'issuetype', 'summary']),
+            ('operation', 'comment', ['issue', 'comment']),
+            ('operation', 'fetch', ['issue']),
+            ('operation', 'transition', ['issue', 'status']),
```
Comment (felixfontein) [inline plugins/modules/web_infrastructure/jira.py:494]: The original code did not require `issue`.

Comment (russoz) [reply plugins/modules/web_infrastructure/jira.py:494]: Yeah it did not, and wrongly so. If you look at the ``transition() `` code you'll see that it does require that for building the URI. And the only thing we can "transition" in Jira is an issue.

Should I explicitly mention this in the changelog fragment?

Comment (felixfontein) [reply plugins/modules/web_infrastructure/jira.py:494]: Goot spot! And yes, that's definitely something that should be mentioned in the fragment.

Comment (russoz) [reply plugins/modules/web_infrastructure/jira.py:494]: done

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
While adding the traceback is a new feature, I think it's ok to backport it.

**Comment by russoz** [comment]
Thanks for the review @felixfontein !

**Pull request merged**